### PR TITLE
Initial StyleCopTester implementation

### LINKED ISSUE #952: Command line tool to run StyleCopAnalyzers on a solution
It would be nice to have a command line utility that runs the StyleCopAnalyzers analyzers on a solution (code fixes considered later).

```
StyleCopTester [options] <Solution>
```

By default, all analyzers which are enabled by default run. The following options are avaliable for `[options]`:

| Option | Meaning |
| --- | --- |
| `/all` | Run all StyleCopAnalyzers analyzers, including ones that are disabled by default |
| `/id:<id>` | Enable analyzer with diagnostic ID `<id>` (when this is specified, only this analyzer is enabled) |

:memo: A primary purpose of this tool is testing performance with both the sampling and instrumenting profilers. The Test Explorer in Visual Studio 2015 does not currently provide enough ability to configure profiling methods for unit tests, and running inside the IDE introduces significant variance due to the editor environment, hence the need to create this as a command line tool.

### PR DESCRIPTION
**Pull request by pdelvo**
This implementation is currently not really usefull. When certain analyzers are included in the list of analyzers that are used it will not find any diagnostics. This bug is fixed in roslyn versions post rc2. Also currently there isnt really much validation of the command line parameters.

Do we want tests for this?

Fixes #952

### CODE REVIEW

File: StyleCop.Analyzers/StyleCopTester/Program.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+                Console.WriteLine($"Found {diagnostics.Count} diagnostics in {stopwatch.ElapsedMilliseconds}ms");
+            }
+        }
+
+        private static IEnumerable<DiagnosticAnalyzer> FilterAnalyzers(IEnumerable<DiagnosticAnalyzer> analyzers, string[] args)
+        {
+            bool useAll = args.Contains("/all");
+
+            HashSet<string> ids = new HashSet<string>(args.Where(y => y.StartsWith("/id:")).Select(y => y.Substring(4)));
```
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCopTester/Program.cs]: Why we need these if useAll is true?

Comment (sharwell) [inline StyleCop.Analyzers/StyleCopTester/Program.cs]: We don't. But it also doesn't really matter if we do calculate them.

File: StyleCop.Analyzers/StyleCopTester/Program.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        {
+            Compilation compilation = await project.GetCompilationAsync().ConfigureAwait(false);
+            CompilationWithAnalyzers compilationWithAnalyzers = compilation.WithAnalyzers(analyzers);
+
+            return await compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync().ConfigureAwait(false);
+        }
+
+        private static void PrintHelp()
+        {
+            Console.WriteLine("Usage: StyleCopTester [options] <Solution>");
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCopTester/Program.cs]: This should be `Console.WriteLine("Usage: StyleCopTester <Solution> [options]")`

Comment (pdelvo) [inline StyleCop.Analyzers/StyleCopTester/Program.cs]: I should make the command line parsing a bit more robust and support both

File: StyleCop.Analyzers/StyleCopTester/Program.cs (hunk 1 above)
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCopTester/Program.cs]: More obviously skip specified diagnostics if both `/all` and `/id` options presented.

    StyleCopTester /all /id:SA1010 /id:SA1011

File: StyleCop.Analyzers/StyleCopTester/Program.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    /// <summary>
+    /// StyleCopTester is a tool that will analyze a solution, find diagnostics in it and will print out the number of
+    /// diagnostics it could find. This is useful to easily test performance without having the overhead of visual
+    /// studio running.
+    /// </summary>
+    internal class Program
+    {
+        private static void Main(string[] args)
+        {
+            // A valid call must have at least two parameters. A solution file and /all or one /id:SAXXXX
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCopTester/Program.cs]: :question: If only one parameter is given (the solution file), I expected it to enable all diagnostics which are enabled by default (just run the default configuration). I don't mind creating a separate issue for this, but I wanted to check and see if you implemented it this way already.

**Comment by sharwell** [comment]
This will be an awesome addition to the project. :sunglasses:

File: StyleCop.Analyzers/StyleCopTester/Program.cs (hunk 3 above)
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCopTester/Program.cs]: I don't do this right now. I would have to do this manually. Im going to do that

**Pull request merged**